megamek/src/megamek/common/templates/TROView.java

```diff
@@ -439,26 +429,30 @@ protected int addEquipment(Entity entity, boolean includeAmmo) {
             if (m.isOmniPodMounted() || !entity.isOmni()) {
                 final String loc = formatLocationTableEntry(entity, m);
                 equipment.putIfAbsent(loc, new HashMap<>());
-                equipment.get(loc).merge(m.getType(), 1, Integer::sum);
+                equipment.get(loc).merge(new EquipmentKey(m.getType(), m.getSize(), m.isArmored()),
+                        1, Integer::sum);
             }
         }
         final List<Map<String, Object>> eqList = new ArrayList<>();
         for (final String loc : equipment.keySet()) {
-            for (final Map.Entry<EquipmentType, Integer> entry : equipment.get(loc).entrySet()) {
-                final EquipmentType eq = entry.getKey();
-                final int count = equipment.get(loc).get(eq);
-                String name = stripNotes(eq.getName());
+            for (final Map.Entry<EquipmentKey, Integer> entry : equipment.get(loc).entrySet()) {
+                final EquipmentType eq = entry.getKey().getType();
+                final int count = equipment.get(loc).get(entry.getKey());
+                String name = stripNotes(entry.getKey().name());
+                if (entry.getKey().isArmored()) {
+                    name += " (Armored)";
+                }
                 if (eq instanceof AmmoType) {
                     name = String.format("%s (%d)", name, ((AmmoType) eq).getShots() * count);
```

Comment:
The name variable appears to get overwritten if the equipment is not ammo but takes up more than one slot?

Reply:
Actually it's if there's more than one of them. It adds the count and appends an 's' to the name. But the formation of a plural is not correct in cases where the equipment name itself ends in 's' and the possible addition of "(Armored)" throws it off in any case. So I removed the grammatical inflection as well as including the armored tag.